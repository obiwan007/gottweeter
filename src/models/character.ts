export class Character {
  // The hypermedia URL of this resource
  public url: string;

  public name: string;
  // The name of this character
  public gender: string;
  // The gender of this character.
  public culture: string;
  // the culture that this character belongs to.
  public born: string;
  // Textual representation of when and where this character was born.
  public died: string;
  // Textual representation of when and where this character died.
  public titles: string[];
  // TThe titles that this character holds.
  public aliases: string[];
  // The aliases that this character goes by.
  public father: string
  // The character resource URL of this character's father.
  public mother: string;
  // The character resource URL of this character's mother.
  public spouse: string;
  // An array of Character resource URLs that has had a POV - chapter in this book.
  public allegiances: string[];
  // An array of House resource URLs that this character is loyal to.
  public books: string[];
  // An array of Book resource URLs that this character has been in.
  public povBooks: string[];
  // An array of Book resource URLs that this character has had a POV - chapter in.
  public tvSeries: string[];
  // An array of names of the seasons of Game of Thrones that this character has been in.
  public playedBy: string[];
}