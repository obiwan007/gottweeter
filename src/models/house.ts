export class House {
  public url: string; // The hypermedia URL of this resource
  public name: string; // The name of this house
  public region: string; // The region that this house resides in.
  public coatOfArms: string; // Text describing the coat of arms of this house.
  public words: string; // The words of this house.
  public titles: string[]; // The titles that this house holds.
  public seats: string[]; // The seats that this house holds.
  public currentLord: string; // The Character resource URL of this house's current lord.
  public heir: string; // The Character resource URL of this house's heir.
  public overlord: string; // The House resource URL that this house answers to.
  public founded: string; // The year that this house was founded.
  public founder: string; // The Character resource URL that founded this house.
  public diedOut: string; // The year that this house died out.
  public ancestralWeapons: string[]; // An array of names of the noteworthy weapons that this house owns.
  public cadetBranches: string[]; // An array of House resource URLs that was founded from this house.
  public swornMembers: string[]; // An array of Character resource URLs that are sworn to this house.
}