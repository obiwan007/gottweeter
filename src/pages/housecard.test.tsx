import * as React from 'react';

import { ShallowRenderer } from 'react-test-renderer/shallow'; // ES6

import { shallow } from 'enzyme';

import HouseCard from "./housecard";

import { House } from '../models/house';

describe("HouseCard", () => {

  it("HouseCard renders correctly if empty", () => {
    const component = shallow(
      <HouseCard
      />,
      { disableLifecycleMethods: false }
    );
    expect(component).toMatchSnapshot();
  });

  it("HouseCard renders correctly with House", () => {
    const house = new House();
    house.name = 'My House';
    const component = shallow(
      <HouseCard
        house = {house}
      />,
      { disableLifecycleMethods: false }
    );
    expect(component).toMatchSnapshot();
  });

});
