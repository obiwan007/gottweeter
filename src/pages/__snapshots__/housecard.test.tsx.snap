// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseCard HouseCard renders correctly if empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(HouseCard) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "bullet": "HouseCard-bullet-1",
        "card": "HouseCard-card-2",
        "icon": "HouseCard-icon-3",
        "pos": "HouseCard-pos-4",
        "progress": "HouseCard-progress-5",
        "root": "HouseCard-root-6",
        "title": "HouseCard-title-7",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "bullet": "HouseCard-bullet-1",
          "card": "HouseCard-card-2",
          "icon": "HouseCard-icon-3",
          "pos": "HouseCard-pos-4",
          "progress": "HouseCard-progress-5",
          "root": "HouseCard-root-6",
          "title": "HouseCard-title-7",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;

exports[`HouseCard HouseCard renders correctly with House 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(HouseCard)
    house={
      House {
        "name": "My House",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "bullet": "HouseCard-bullet-1",
        "card": "HouseCard-card-2",
        "icon": "HouseCard-icon-3",
        "pos": "HouseCard-pos-4",
        "progress": "HouseCard-progress-5",
        "root": "HouseCard-root-6",
        "title": "HouseCard-title-7",
      },
      "house": House {
        "name": "My House",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "bullet": "HouseCard-bullet-1",
          "card": "HouseCard-card-2",
          "icon": "HouseCard-icon-3",
          "pos": "HouseCard-pos-4",
          "progress": "HouseCard-progress-5",
          "root": "HouseCard-root-6",
          "title": "HouseCard-title-7",
        },
        "house": House {
          "name": "My House",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;
